### PR TITLE
Fixes

### LINKED ISSUE #8: [BAN] 400 Bad Request response have no body
When we sent request **POST /bans** with one of next body:

> {
            "target": [{
                "user_agent": "user-agent for array test"
            }],
            "reason": "Ban through IP",
            "ttl": 321
        }

or 

> [
            {
                "target": {
                    "user_agent": "user-agent for array test"
                },
                "reason": "Ban through IP",
                "ttl": 321
            }
        ]

response had no body

### LINKED ISSUE #9: [BAN] Unexpected response for request with wrong user-agent field
I sent POST /bans request with **user-agen** instead of **user_agent** field:

> {
            "target": {
                "ip": "123.123.123.123",
                "user-agent": "user-agent for array test"
            },
            "reason": "Ban through IP",
            "ttl": 322
        }

but I had 204 Ok response without any warnings or errors. In Redis has been created key which contains ip field only.

### LINKED ISSUE #10: [BAN] Can't ban user throug user-agent only
When I sent request with next body:

> {
    "target": {
        "user_agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36"
    },
    "reason": "Ban through User-Agent",
    "ttl": 4294967297
}

I awaited a 204 response, but had a 400 Bad Request without response body

### LINKED ISSUE #11: [BAN] Too big ttl 400 response without body
On the request **POST /api/bans** when I set ttl equal or bigger than 2^32 like this:

> {
    "target": {
        "ip": "12.34.56.78"
    },
    "reason": "Ban through User-Agent",
    "ttl": 4294967296
}

I get 400 Bad Request response with nothing body

### LINKED ISSUE #15: [BAN] Empty error response for request with broken IP target
When I send request with numeric target IP field:

> {
    "target": {
        "ip": 192
    },
    "reason": "Strange ban with empty user-agent and no-empty IP",
    "ttl": 302
}

I came an empty response body (but response code 400 returns correctly)

### LINKED ISSUE #16: [BAN] Posible to send empty X-Analyzer-Id header
I can send request with empty X-Analyzer-Id header like this:

> {"X-Analyzer-Id": ""}

and I receive 204 response, and created hashset record in Redis with empty key.

### PR DESCRIPTION
**Pull request by horny-manul**
Closes #8 
Closes #9
Closes #10 
Closes #11
Closes #15
Closes #16

Closes https://github.com/crypto-pepe/firewall-executor/issues/14
Closes https://github.com/crypto-pepe/firewall-executor/issues/13
Closes https://github.com/crypto-pepe/firewall-executor/issues/12

### CODE REVIEW

File: src/api/server.rs [hunk 1]
```diff
@@ -47,7 +48,17 @@ fn server_config() -> Box<dyn Fn(&mut web::ServiceConfig)> {
         let json_cfg = web::JsonConfig::default()
             .content_type(|mime| mime == mime::APPLICATION_JSON)
             .error_handler(|err, _| {
-                error::InternalError::from_response(err, HttpResponse::BadRequest().into()).into()
+                let reason = err.to_string();
+                error::InternalError::from_response(
+                    err,
+                    ErrorResponse {
+                        code: 400,
```
Comment (citizensix) [inline src/api/server.rs:55]: ```use StatusCode::...```

File: src/api/routes/ban.rs [hunk 2]
```diff
@@ -45,7 +45,16 @@ pub async fn process_ban(
             });
         }
         Some(s) => match s.to_str() {
-            Ok(s) => s,
+            Ok(s) => match s {
+                "" => {
+                    return Err(ErrorResponse {
+                        code: 400,
```
Comment (citizensix) [inline src/api/routes/ban.rs:51]: StatusCode::BAD_REQUEST.as_u16()

File: src/api/routes/ban.rs [hunk 3]
```diff
@@ -45,7 +45,16 @@ pub async fn process_ban(
             });
         }
         Some(s) => match s.to_str() {
-            Ok(s) => s,
+            Ok(s) => match s {
+                "" => {
```
Comment (citizensix) [inline src/api/routes/ban.rs:49]: `s.is_empty`

File: src/api/routes/ban.rs [hunk 4]
```diff
@@ -45,7 +45,16 @@ pub async fn process_ban(
             });
         }
         Some(s) => match s.to_str() {
```
Comment (citizensix) [inline src/api/routes/ban.rs:47]: You can use `map_or()` in this case

**Review by citizensix — COMMENTED** [review]
.

**Review by stuffy-idler — CHANGES_REQUESTED** [review]

**Comment by stuffy-idler** [comment]
@dummy-duck please check whether issues were resolved

**Review by stuffy-idler — CHANGES_REQUESTED** [review]

**Review by dummy-duck — CHANGES_REQUESTED** [review]
1) Wrong Dockerfile - image can't be builded
2) https://github.com/crypto-pepe/firewall-executor/issues/10 - not solved
3) https://github.com/crypto-pepe/firewall-executor/issues/9 - not solved

**Comment by horny-manul** [comment]
@dummy-duck 

> https://github.com/crypto-pepe/firewall-executor/issues/10 - not solved

In your example request ttl is 4294967297, which is not uint32 therefore it fails. 

> https://github.com/crypto-pepe/firewall-executor/issues/9 - not solved

Fixed. Now all unknown fields produces an error

**Comment by stuffy-idler** [comment]
@horny-manul
> In your example request ttl is 4294967297, which is not uint32 therefore it fails.

Will server response with bad request for this case? It has to

File: src/api/http_error.rs [hunk 5]
```diff
@@ -35,6 +37,30 @@ impl From<BanTargetConversionError> for ErrorResponse {
...
+            code: 400,
+            reason: e.to_string(),
+            details: None,
+        }
+    }
+}
+
 impl Display for ErrorResponse {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         f.write_str(serde_json::to_string(self).unwrap().as_str())
```
Comment (stuffy-idler) [inline src/api/http_error.rs:66]: Has not to panic. 
Let's get rid of `unwrap`, change it to `Err`

File: src/api/routes/ban.rs [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
-use crate::api::http_error::ErrorResponse;
+use crate::api::http_error::{ErrorResponse, HeaderError};
 use crate::api::ANALYZER_HEADER;
 use crate::ban_hammer::BanHammerDryRunner;
 use crate::model::{BanEntity, BanTarget};
 
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(rename_all = "snake_case")]
+#[serde(deny_unknown_fields)]
 pub struct BanRequest {
```
Comment (stuffy-idler) [inline src/api/routes/ban.rs:16]: `Display` impl has not to panic (`expect` usage). Please fix it – impl is fallible, therefore it can be changed to `Err`

**Review by stuffy-idler — CHANGES_REQUESTED** [review]

**Review by stuffy-idler — APPROVED** [review]

**Pull request merged**